src/abstractdataframe/join.jl

```diff
@@ -280,7 +286,22 @@ join(name, job2, on = :ID => :identifier)
 function Base.join(df1::AbstractDataFrame,
                    df2::AbstractDataFrame;
                    on::Union{<:OnType, AbstractVector{<:OnType}} = Symbol[],
-                   kind::Symbol = :inner, makeunique::Bool=false)
+                   kind::Symbol = :inner, makeunique::Bool=false,
+                   indicator::Union{Nothing, Symbol} = nothing)
+
+    if !(indicator === nothing)
+        names_in_use = Set([names(df1); names(df2); [indicator]])
```

Comment:
The `DataFrame` type already includes a dictionary of column names. Just do `c =  Symbol(indicator_cols[i]); haskey(index(df1), c) || haskey(index(df2), c) || c == indicator` in the loop.